File: longestCommonPrefix.js
class Solution {
  public String longestCommonPrefix(String[] strs) {
        String first = strs[0];
    if (first.length() == 0) {
      return "";
    }
    for (int i = 1; i < strs.length; i++) {
      if (strs[i] == "") {
        return "";
      }
      if (strs[i].length() == 1 && strs[i].charAt(0) == first.charAt(0)) {
        first = String.valueOf(first.charAt(0));
      } else if (strs[i].length() == 1 && strs[i].charAt(0) != first.charAt(0)) {
        return "";
      } else {
        for (int j = 0; j < strs[i].length(); j++) {
          if (j >= first.length() || strs[i].charAt(j) != first.charAt(j)) {
            first = first.substring(0, j);
            break;
          }
        }
      }
    }
    return first;
  }
}

File: longestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String result = strs[0];
        String update = "";
        if (result.length() == 0) {
            return "";
        }
        for (int i = 1; i < strs.length; i++) {
            for (int j = 0; j < strs[i].length(); j++) {
                if (result.length() < j + 1) {
                    break;
                }
                if (strs[i].charAt(j) == result.charAt(j)) {
                    update+=strs[i].charAt(j);
                } else {
                    break;
                }
            }
            result = update;
            update = "";
        }
        return result;
    }
}


class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        String result = strs[0];
        for (int i = 1; i < strs.length; i++) {
            while (strs[i].indexOf(result) != 0) {
                result = result.substring(0, result.length() - 1);
                if (result.isEmpty()) {
                    return "";
                }
            }
        }
        return result;
    }
}
